# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: editor/libeditor/TextEditSubActionHandler.cpp
# Commit: 71670cedaf26
# Full Hash: 71670cedaf26214f3bcb8562a445cf1ef24f4db6
# Author: Masayuki Nakano <masayuki@d-toybox.com>
# Date: 2021-06-30 09:36:48
# Regressor Bug: 1714914
# File Overlap Count: 1
# Description:
#   Bug 1714914 - Make editor classes treat offset in parent node as uint32_t r=m_kato
#   
#   It should be treated as `uint32_t` since DOM API does so.  However, there are
#   some exceptions:
#   
# ==============================================================================

diff -r f5dcc2d793d6 -r 71670cedaf26 editor/libeditor/TextEditSubActionHandler.cpp
--- a/editor/libeditor/TextEditSubActionHandler.cpp	Tue Jun 29 22:00:09 2021 +0000
+++ b/editor/libeditor/TextEditSubActionHandler.cpp	Wed Jun 30 07:07:28 2021 +0000
@@ -721,7 +721,7 @@
       break;
   }
 
-  int32_t currentLength = INT32_MAX;
+  uint32_t currentLength = UINT32_MAX;
   nsresult rv = GetTextLength(&currentLength);
   if (NS_FAILED(rv)) {
     NS_WARNING("TextEditor::GetTextLength() failed");
@@ -742,13 +742,13 @@
   //     is part of kOldCompositionStringLength.
   const uint32_t kNewLength =
       currentLength - kSelectionLength - kOldCompositionStringLength;
-  if (kNewLength >= static_cast<uint32_t>(mMaxTextLength)) {
+  if (kNewLength >= AssertedCast<uint32_t>(mMaxTextLength)) {
     aInsertionString.Truncate();  // Too long, we cannot accept new character.
     return EditActionHandled();
   }
 
   if (aInsertionString.Length() + kNewLength <=
-      static_cast<uint32_t>(mMaxTextLength)) {
+      AssertedCast<uint32_t>(mMaxTextLength)) {
     return EditActionIgnored();  // Enough short string.
   }
 